src/server/pps/server/api_server.go

```diff
@@ -180,6 +182,10 @@ func validateNames(names map[string]bool, input *pps.Input) error {
 }
 
 func (a *apiServer) validateInput(pachClient *client.APIClient, pipelineName string, input *pps.Input, job bool) error {
+	if !pipelineNameMatcher.MatchString(pipelineName) {
+		return fmt.Errorf("Invalid pipeline name: a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345')")
```

Comment:
Referring to this as a `label` is probably going to be confusing to users who get it back. Also we should probably disallow empty strings for pipeline names, I can't think of a real use case for that.